src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -90,6 +90,7 @@ public Server(ClusterId clusterId, IClusterClock clusterClock, ClusterConnection
             _connectionPool = connectionPoolFactory.CreateConnectionPool(_serverId, endPoint);
             _state = new InterlockedInt32(State.Initial);
             _monitor = serverMonitorFactory.Create(_serverId, _endPoint);
+            _baseDescription = _currentDescription = _monitor.Description;
```

Comment:
I've split up the assignments! (Apologies, I initially was copying the double assignment in a single line pattern from `ServerMonitor`).

I like your idea of not relying on the monitor description and have refactor the init of  `_baseDescription` so that we're not using the monitor's description.